README.md
# Capernia Datapack: making the ultimate datapack for OTS (Open Tibia Server)[![Build Status](https://img.shields.io/badge/build-buggy-yellow.svg)]()


Capernia Datapack is a project born from the necessity of a serious OTS files.
Its goal is to make sure all new Tibia resources be unified and ready for use on the latest version of Forgotten Server.


![Tibia 10.91 Sample](https://cloud.githubusercontent.com/assets/19177793/15003885/258236d8-1187-11e6-86fd-f83118992146.png)


Capernia Datapack is made using **LUA scripting** language with some **mysql arguments**.

![Latest version!](https://img.shields.io/badge/tfs%20version-1.2-brightgreen.svg)
![There are newer versions!](https://img.shields.io/badge/client%20version-10.93-green.svg)
[![Gitter](https://badges.gitter.im/gordonbay/capernia-ots-datapack.svg)](https://gitter.im/gordonbay/capernia-ots-datapack?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)


Main Conceps
=============

* Analyse and scrap third party datapacks for new userful content
* Update the scripts making them compatible with latest version of Forgotten Server
* Unify all existing datapacks on Capernia datapack
* Help the community



Installation
============


On **Windows**, just extract all contents, change variables on **config.lua** to match your MYSQL settings, import MYSQL database (*.sql) and run the provided binary.

 - Unpack global.rar to same folder:

    > capernia-ots-datapack\data\world\global.rar

 - Edit config.lua settings:
![Lua script](https://img.shields.io/badge/script-lua-blue.svg)
	```lua
	mysqlHost = "127.0.0.1" 
	mysqlUser = "root"
	mysqlPass = "az88231015"
	mysqlDatabase = "servidor"
	```
 - Import `schema.sql`to your mysql database:

    > capernia-ots-datapack\res\schema.sql
    
 - Have fun!


Github Limitations
===========

Because of the `size` limits, the following file had to be compressed.

    capernia-ots-datapack\data\world\global.rar

Map Changes
===========

For obvious reasons we have a diferent protocol for changing the map.

 - Open a issue, using the `Issues` tab
 - Explain the changes needed and if possible, post an image

Datapacks Merges
=============

This list shows the status of third party datapacks **migrations**:

| Name | Last Author Release | Last Capernia Merge | Capernia Merge Status | Location | Report |
| --- | --- | --- | --- | --- | --- |
| Mitisuig | 13/04/2016 | 13/04/2016 | ![Complete!](https://img.shields.io/badge/status-complete-brightgreen.svg) | [TK](http://www.tibiaking.com/forum/topic/36403-global-full-tibiaking-10911092-tfs-12-full-oramond-cast-system-lions-rock-reward-system-equip-hotkey-new-items-news-monster-new-areas-krailos-asura-medusa-town-etc-eventos-sem-missao-v20/) | [Read Report](https://github.com/gordonbay/capernia-ots-datapack/issues/18) |
| Absolute | 03/05/2016 | - | ![Aware of this but nothing done yet!](https://img.shields.io/badge/aware-0%25-red.svg) | [TK](http://www.tibiaking.com/forum/topic/67075-global-full-download-1090-tfs-12ferumbras-questkrailosnew-arenacastrewardeventscasino/) | ![](https://img.shields.io/badge/invitation-pending-red.svg) |
| Lofiv | 02/03/2016 | - | ![Aware of this but nothing done yet!](https://img.shields.io/badge/aware-0%25-red.svg) | [OTLand](https://otland.net/threads/10-90-globalot-full-new-www-new-layout-krailos-tfs-1-2.241083/) | ![](https://img.shields.io/badge/invitation-pending-red.svg) |


Important Wiki Pages
=============

- [capernia-ots-datapack/wiki/Issues](https://github.com/gordonbay/capernia-ots-datapack/wiki/Issues) - Read this before report any issue


CHANGELOG.md
## 0.2.0 (01/07/2016)

Bugfixes:
  - fix `potions.lua` exhaust time as requested [here #44](https://github.com/gordonbay/capernia-ots-datapack/issues/44)

## 0.1.9 (07/06/2016)

Bugfixes:
  - fix problems with `weapons.xml` file as requested [here #40](https://github.com/gordonbay/capernia-ots-datapack/issues/40)


## 0.1.8 (04/06/2016)

Bugfixes:
  - fix wrong items values at `items.xml`
  - fix wrong textures and missing boat at `33358, 31588, 12` `33356, 31589, 11` `33365, 31612, 11` `33677, 31598, 7`
  - fix `Renegade Knight` drops and stats


## 0.1.7 (30/05/2016)

Bugfixes:
  - fix missing or wrong `items.xml` values that could make client crash. Hive structure, gooey substance, aniversary cake, stairs, adventurer stone, strange growth, insectoid cocoon, hive structure, insectoid hivecomb, dry insectoid hivecomb, insectoid eggs, cracked insectoid eggs, hive gate, cup of molten gold, tarantula trap, trapdoor, rolling pin, peeing orc, fire, destroy field rune, pitch black gap, hole, bog water, star ring, pallet, trunk, leaves, mountain, clay ground, the dead keeper, fire basin, table, dirt floor, rutted stone, jade ornaments, dragon claw lamp, iron case, putrid growth, acid, large black crystal, giant dreadcoil, doomato, witherstem, carnivortex, toxic tulip, clay lump, cursed shoulder spikes and others...
  - fix missing `schema.sql` file as requested [here #33](https://github.com/gordonbay/capernia-ots-datapack/issues/33)
  - fix `enchantment system` as requested [here #32](https://github.com/gordonbay/capernia-ots-datapack/issues/32)
  
  Enhancements:
  - fix wrong NPC `Yaman, Haroun and Grizzly Adams` coordinates as requested [here #32](https://github.com/gordonbay/capernia-ots-datapack/issues/32)

  Additions:
  - `Gesior` php data is now supported by this version

## 0.1.6 (27/05/2016)

Bugfixes:
  - fix `The Explorer Society Quest` errors as requested [here #27](https://github.com/gordonbay/capernia-ots-datapack/issues/27) 

## 0.1.5 (25/05/2016)

Bugfixes:
  - fix `spellbook` errors as requested [here #25](https://github.com/gordonbay/capernia-ots-datapack/issues/25) 

## 0.1.4 (23/05/2016)

Enhancements:
  - fix some `map` glitches
  - some changes on `Killing in the Name of...` necromancer task as partially requested [here #22](https://github.com/gordonbay/capernia-ots-datapack/issues/22) 
  - changed the compiled executables, now it comes both x64 and x86 versions and with respective dll

## 0.1.3 (20/05/2016)

Bugfixes:
  - fix `The New Frontier Quest` teleport not checking player storage as requested [here #21](https://github.com/gordonbay/capernia-ots-datapack/issues/21) 

## 0.1.2 (19/05/2016)

Bugfixes:
  - fix `surprise bags` errors as requested [here #20](https://github.com/gordonbay/capernia-ots-datapack/issues/20) 

## 0.1.1 (18/05/2016)

Enhancements:
  - fix wrong `spells` formulas as requested [here #19](https://github.com/gordonbay/capernia-ots-datapack/issues/19) 

## 0.1.0 (17/05/2016)

Bugfixes:
  - fix missing `monsters spells`

## 0.0.9 (17/05/2016)

Bugfixes:
  - fix wrong monsters `corpses` as requested [here #17](https://github.com/gordonbay/capernia-ots-datapack/issues/17)
  - fix missing `Yalahar` gates and `Captain Max` not checking mission status in order to teleport a character. As described [here #16](https://github.com/gordonbay/capernia-ots-datapack/issues/16)
  - fix Edron's `Bog Raider's` missing spaws and wrong map textures as requested [here #14](https://github.com/gordonbay/capernia-ots-datapack/issues/14)

## 0.0.8 (14/05/2016)

Enhancements:
  - added `10.92` new itens and potions

## 0.0.7 (13/05/2016)

Bugfixes:
  - fix a bug on `Ferumbras' Ascendant Quest` as requested [here #13](https://github.com/gordonbay/capernia-ots-datapack/issues/13)

Enhancements:
  - `Reward Chest` has been removed from Thais due violations of Capernia rules about custom content


## 0.0.6 (12/05/2016)

Bugfixes:
  - fix errors on NPC `Chemar`
  - fix incorrect textures on `Darashia` -9 level, at rotworms cave
  
Enhancements:
  - NPC Chemar now respect `The New Frontier` quest in order to take you to `Farmine`


## 0.0.5 (11/05/2016)

Bugfixes:
  - fix melee weapons not being able to cut webs on the `Cruzader Helmet Quest`
  
Enhancements:
  - melee weapons now break or destroy some itens
  - spider eggs should work properly now

## 0.0.4 (10/05/2016)

Additions:
  - added missing `spaws` around theese spots:
    - 33351,31706,13
    - 33351,31678,13
    - 33351,31671,13
    - 33330,31692,12
    - 33337,31650,12
    - 33351,31671,13
    - 32987,31629,12
    - 33190,32056,11
    - 33190,32042,11
    - 33183,32035,11
    - 33162,32000,11
    - 33204,31986,11
    - 33365,31692,11
    - 33379,31671,11
    - 33414,31650,11
    - 33197,32434,9
    - 33211,32441,9
    - 33204,32413,9
    - 33337,31993,9
    - 33337,31979,9
    - 33001,31678,9
    - 33008,31650,9
    - 33576,31599,9
    - 32652,33007,8
    - 32593,32994,8
    - 32659,32983,8
    - 32619,32978,8
    - 32154,32868,8
    - 33330,31979,8
    - 33316,31965,8
    - 33092,31713,8
    - 32154,32868,7
    - 32854,32798,7
    - 33176,32469,7
    - 32154,32861,6
    - 33505,31608,7
  
Enhancements:
  - added an entire missing island underground at 33188,32988,7
  - fixed glitchs on the map file
  - removed Thais trainers because of the new Capernia rules on custom stuff

## 0.0.3 (09/05/2016)

Additions:
  - added missing NPC `Ulala`
  - added missing NPC `A Dark Priestess`
  - added missing NPC `Rapanaio`
  - added missing NPC `Mother Of Jack`
  - added missing NPC `Pythius The Rotten`
  - added missing NPC `Awarness Of The Emperor`
  - added missing NPC `A Ghostly Woman`
  - added missing NPC `Junkar`
  - added missing NPC `The Queen Of The Banshees`
  - added missing NPC `Grombur`
  - added missing NPC `Frafnar`
  - added missing NPC `Nokmir`
  - added missing NPC `Robson`
  - added missing NPC `A Wrinkled Bonelord`
  - added missing NPC `Shortsighted Dwarf`
  - added missing NPC `A Tortured Soul`

data/npc/scripts/Rashid.lua
local keywordHandler = KeywordHandler:new()
local npcHandler = NpcHandler:new(keywordHandler)
NpcSystem.parseParameters(npcHandler)

function onCreatureAppear(cid)			npcHandler:onCreatureAppear(cid)			end
function onCreatureDisappear(cid)		npcHandler:onCreatureDisappear(cid)			end
function onCreatureSay(cid, type, msg)		npcHandler:onCreatureSay(cid, type, msg)		end
function onThink()				npcHandler:onThink()					end

local function creatureSayCallback(cid, type, msg)
	if(not npcHandler:isFocused(cid)) then
		return false
	end
	local player = Player(cid)
	if(msgcontains(msg, "mission")) then
		if(os.date("%A") == "Monday") then
			if(player:getStorageValue(Storage.TravellingTrader.Mission01) < 1) then
				npcHandler:say("Well, you could attempt the mission to become a recognised trader, but it requires a lot of travelling. Are you willing to try?", cid)
				npcHandler.topic[cid] = 1
			elseif(player:getStorageValue(Storage.TravellingTrader.Mission01) == 1) then
				npcHandler:say("Have you managed to obtain a rare deer trophy for my customer?", cid)
				npcHandler.topic[cid] = 3
			end
		elseif(os.date("%A") == "Tuesday") then
			if(player:getStorageValue(Storage.TravellingTrader.Mission01) == 2 and player:getStorageValue(Storage.TravellingTrader.Mission02) < 1 ) then
				npcHandler:say("So, my friend, are you willing to proceed to the next mission to become a recognised trader?", cid)
				npcHandler.topic[cid] = 4
			elseif(player:getStorageValue(Storage.TravellingTrader.Mission02) == 4) then
				npcHandler:say("Did you bring me the package?", cid)
				npcHandler.topic[cid] = 6
			end
		elseif(os.date("%A") == "Wednesday") then
			if(player:getStorageValue(Storage.TravellingTrader.Mission02) == 5 and player:getStorageValue(Storage.TravellingTrader.Mission03) < 1 ) then
				npcHandler:say("So, my friend, are you willing to proceed to the next mission to become a recognised trader?", cid)
				npcHandler.topic[cid] = 7
			elseif(player:getStorageValue(Storage.TravellingTrader.Mission03) == 2) then
				npcHandler:say("Have you brought the cheese?", cid)
				npcHandler.topic[cid] = 9
			end
		elseif(os.date("%A") == "Thursday") then
			if(player:getStorageValue(Storage.TravellingTrader.Mission03) == 3 and player:getStorageValue(Storage.TravellingTrader.Mission04) < 1) then
				npcHandler:say("So, my friend, are you willing to proceed to the next mission to become a recognised trader?", cid)
				npcHandler.topic[cid] = 10
			elseif(player:getStorageValue(Storage.TravellingTrader.Mission04) == 2) then
				npcHandler:say("Have you brought the vase?", cid)
				npcHandler.topic[cid] = 12
			end
		elseif(os.date("%A") == "Friday") then
			if(player:getStorageValue(Storage.TravellingTrader.Mission04) == 3 and player:getStorageValue(Storage.TravellingTrader.Mission05) < 1) then
				npcHandler:say("So, my friend, are you willing to proceed to the next mission to become a recognised trader?", cid)
				npcHandler.topic[cid] = 13
			elseif(player:getStorageValue(Storage.TravellingTrader.Mission05) == 2) then
				npcHandler:say("Have you brought a cheap but good crimson sword?", cid)
				npcHandler.topic[cid] = 15
			end
		elseif(os.date("%A") == "Saturday") then
			if(player:getStorageValue(Storage.TravellingTrader.Mission05) == 3 and player:getStorageValue(Storage.TravellingTrader.Mission06) < 1) then
				npcHandler:say("So, my friend, are you willing to proceed to the next mission to become a recognised trader?", cid)
				npcHandler.topic[cid] = 16
			elseif(player:getStorageValue(Storage.TravellingTrader.Mission06) == 1) then
				npcHandler:say("Have you brought me a gold fish??", cid)
				npcHandler.topic[cid] = 18
			end
		elseif(os.date("%A") == "Sunday") then
			if(player:getStorageValue(Storage.TravellingTrader.Mission06) == 2 and player:getStorageValue(Storage.TravellingTrader.Mission07) ~= 1) then
				npcHandler:say("Ah, right. <ahem> I hereby declare you - one of my recognised traders! Feel free to offer me your wares!", cid)
				player:setStorageValue(Storage.TravellingTrader.Mission07, 1)
				player:addAchievement('Recognised Trader')
				npcHandler.topic[cid] = 0
			end
		end
	elseif(msgcontains(msg, "yes")) then
		if(npcHandler.topic[cid] == 1) then
			npcHandler:say({
				"Very good! I need talented people who are able to handle my wares with care, find good offers and the like, so I'm going to test you. ...",
				"First, I'd like to see if you can dig up rare wares. Something like a ... mastermind shield! ...",
				"Haha, just kidding, fooled you there, didn't I? Always control your nerves, that's quite important during bargaining. ...",
				"Okay, all I want from you is one of these rare deer trophies. I have a customer here in Svargrond who ordered one, so I'd like you to deliver it tome while I'm in Svargrond. ...",
				"Everything clear and understood?"
			}, cid)

			npcHandler.topic[cid] = 2
		elseif(npcHandler.topic[cid] == 2) then
			npcHandler:say("Fine. Then get a hold of that deer trophy and bring it to me while I'm in Svargrond. Just ask me about your mission.", cid)
			player:setStorageValue(Storage.TravellingTrader.Mission01, 1)
			npcHandler.topic[cid] = 0
		elseif(npcHandler.topic[cid] == 3) then
			if player:removeItem(7397, 1) then
				npcHandler:say("Well done! I'll take that from you. <snags it> Come see me another day, I'll be busy for a while now. ", cid)
				player:setStorageValue(Storage.TravellingTrader.Mission01, 2)
				npcHandler.topic[cid] = 0
			end
		elseif(npcHandler.topic[cid] == 4) then
			npcHandler:say({
				"Alright, that's good to hear. From you as my trader and deliveryman, I expect more than finding rare items. ...",
				"You also need to be able to transport heavy wares, weaklings won't get far here. I have ordered a special package from Edron. ...",
				"Pick it up from Willard and bring it back to me while I'm in Liberty Bay. Everything clear and understood?"
			}, cid)
			npcHandler.topic[cid] = 5
		elseif(npcHandler.topic[cid] == 5) then
			npcHandler:say("Fine. Then off you go, just ask Willard about the 'package for Rashid'.", cid)
			player:setStorageValue(Storage.TravellingTrader.Mission02, 1)
			npcHandler.topic[cid] = 0
		elseif(npcHandler.topic[cid] == 6) then
			if player:removeItem(7503, 1) then
				npcHandler:say("Great. Just place it over there - yes, thanks, that's it. Come see me another day, I'll be busy for a while now. ", cid)
				player:setStorageValue(Storage.TravellingTrader.Mission02, 5)
				npcHandler.topic[cid] = 0
			end
		elseif(npcHandler.topic[cid] == 7) then
			npcHandler:say({
				"Well, that's good to hear. From you as my trader and deliveryman, I expect more than carrying heavy packages. ...",
				"You also need to be fast and deliver wares in time. I have ordered a very special cheese wheel made from Darashian milk. ...",
				"Unfortunately, the high temperature in the desert makes it rot really fast, so it must not stay in the sun for too long. ...",
				"I'm also afraid that you might not be able to use ships because of the smell of the cheese. ...",
				"Please get the cheese from Miraia and bring it to me while I'm in Port Hope. Everything clear and understood?"
			}, cid)
			npcHandler.topic[cid] = 8
		elseif(npcHandler.topic[cid] == 8) then
			npcHandler:say("Okay, then please find Miraia in Darashia and ask her about the {'scarab cheese'}.", cid)
			player:setStorageValue(Storage.TravellingTrader.Mission03, 1)
			npcHandler.topic[cid] = 0
		elseif(npcHandler.topic[cid] == 9) then
			if player:removeItem(8112, 1) then
				npcHandler:say("Mmmhh, the lovely odeur of scarab cheese! I really can't understand why most people can't stand it. Thanks, well done! ", cid)
				player:setStorageValue(Storage.TravellingTrader.Mission03, 3)
				npcHandler.topic[cid] = 0
			end
		elseif(npcHandler.topic[cid] == 10) then
			npcHandler:say({
				"Well, that's good to hear. From you as my trader and deliveryman, I expect more than bringing stinky cheese. ...",
				"I wonder if you are able to deliver goods so fragile they almost break when looked at. ...",
				"I have ordered a special elven vase from Briasol in Ab'Dendriel. Get it from him and don't even touch it, just bring it to me while I'm in Ankrahmun. Everything clear and understood?"
			}, cid)
			npcHandler.topic[cid] = 11
		elseif(npcHandler.topic[cid] == 11) then
			npcHandler:say("Okay, then please find {Briasol} in {Ab'Dendriel} and ask for a {'fine vase'}.", cid)
			player:setStorageValue(Storage.TravellingTrader.Mission04, 1)
			player:addMoney(1000)
			npcHandler.topic[cid] = 0
		elseif(npcHandler.topic[cid] == 12) then
			if player:removeItem(8760, 1) then
				npcHandler:say("I'm surprised that you managed to bring this vase without a single crack. That was what I needed to know, thank you. ", cid)
				player:setStorageValue(Storage.TravellingTrader.Mission04, 3)
				npcHandler.topic[cid] = 0
			end
		elseif(npcHandler.topic[cid] == 13) then
			npcHandler:say({
				"Fine! There's one more skill that I need to test and which is cruicial for a successful trader. ...",
				"Of course you must be able to haggle, else you won't survive long in this business. To make things as hard as possible for you, I have the perfect trade partner for you. ...",
				"Dwarves are said to be the most stubborn of all traders. Travel to {Kazordoon} and try to get the smith {Uzgod} to sell a {crimson sword} to you. ...",
				"Of course, it has to be cheap. Don't come back with anything more expensive than 400 gold. ...",
				"And the quality must not suffer, of course! Everything clear and understood?",
				"Dwarves are said to be the most stubborn of all traders. Travel to Kazordoon and try to get the smith Uzgod to sell a crimson sword to you. ..."
			}, cid)
			npcHandler.topic[cid] = 14
		elseif(npcHandler.topic[cid] == 14) then
			npcHandler:say("Okay, I'm curious how you will do with {Uzgod}. Good luck!", cid)
			player:setStorageValue(Storage.TravellingTrader.Mission05, 1)
			npcHandler.topic[cid] = 0
		elseif(npcHandler.topic[cid] == 15) then
			if player:removeItem(7385, 1) then
				npcHandler:say("Ha! You are clever indeed, well done! I'll take this from you. Come see me tomorrow, I think we two might get into business after all.", cid)
				player:setStorageValue(Storage.TravellingTrader.Mission05, 3)
				npcHandler.topic[cid] = 0
			end
		elseif(npcHandler.topic[cid] == 16) then
			npcHandler:say({
				"My friend, it seems you have already learnt a lot about the art of trading. I think you are more than worthy to become a recognised trader. ...",
				"There is just one little favour that I would ask from you... something personal, actually, forgive my boldness. ...",
				"I have always dreamed to have a small pet, one that I could take with me and which wouldn't cause problems. ...",
				"Could you - just maybe - bring me a small goldfish in a bowl? I know that you would be able to get one, wouldn't you?"
			}, cid)
			npcHandler.topic[cid] = 17
		elseif(npcHandler.topic[cid] == 17) then
			npcHandler:say("Thanks so much! I'll be waiting eagerly for your return then.", cid)
			player:setStorageValue(Storage.TravellingTrader.Mission06, 1)
			npcHandler.topic[cid] = 0
		elseif(npcHandler.topic[cid] == 18) then
			if player:removeItem(8766, 1) then
				npcHandler:say("Thank you!! Ah, this makes my day! I'll take the rest of the day off to get to know this little guy. Come see me tomorrow, if you like.", cid)
				player:setStorageValue(Storage.TravellingTrader.Mission06, 2)
				npcHandler.topic[cid] = 0
			end
		end
	end
	return true
end

keywordHandler:addKeyword({"job"}, StdModule.say, {npcHandler = npcHandler, text = "I am a travelling trader. I don't buy everything, though. And not from everyone, for that matter."})
keywordHandler:addKeyword({"name"}, StdModule.say, {npcHandler = npcHandler, text = "I am Rashid, son of the desert."})
keywordHandler:addKeyword({"offers"}, StdModule.say, {npcHandler = npcHandler, text = "Of course, old friend. You can also browse only armor, legs, shields, helmets, boots, weapons, enchanted weapons, jewelry or miscellaneous stuff."})
keywordHandler:addKeyword({"ab'dendriel"}, StdModule.say, {npcHandler = npcHandler, text = "Elves... I don't really trust them. All this talk about nature and flowers and treehugging... I'm sure there's some wicked scheme behind all this."})
keywordHandler:addKeyword({"desert"}, StdModule.say, {npcHandler = npcHandler, text = "My beloved hometown! Ah, the sweet scent of the desert sands, the perfect shape of the pyramids... stunningly beautiful."})
keywordHandler:addKeyword({"karmia"}, StdModule.say, {npcHandler = npcHandler, text = "I have to go to Karmia once in a while, since the queen wishes to see my exclusive wares in regular intervals."})
keywordHandler:addKeyword({"cormaya"}, StdModule.say, {npcHandler = npcHandler, text = "Cormaya? Not a good place to make business, it's way too far and small."})
keywordHandler:addKeyword({"darashia"}, StdModule.say, {npcHandler = npcHandler, text = "It's not the real thing, but almost as good. The merchants there claim ridiculous prices, which is fine for my own business."})
keywordHandler:addKeyword({"premy island"}, StdModule.say, {npcHandler = npcHandler, text = "Ah yes, Premy Island! Such a lovely and quiet island! I usually make some nice business there."})
keywordHandler:addKeyword({"bibula"}, StdModule.say, {npcHandler = npcHandler, text = "Too few customers there, it's not worth the trip."})
keywordHandler:addKeyword({"greenshore"}, StdModule.say, {npcHandler = npcHandler, text = "Um... I don't think so."})
keywordHandler:addKeyword({"moria"}, StdModule.say, {npcHandler = npcHandler, text = "I don't like being underground much. I also tend to get lost in these labyrinthine dwarven tunnels, so I rather avoid them."})
keywordHandler:addKeyword({"blood bay"}, StdModule.say, {npcHandler = npcHandler, text = "When you avoid the slums, it's a really pretty city. Almost as pretty as the governor's daughter."})
keywordHandler:addKeyword({"northport"}, StdModule.say, {npcHandler = npcHandler, text = "Um... I don't think so."})
keywordHandler:addKeyword({"port hope"}, StdModule.say, {npcHandler = npcHandler, text = "I like the settlement itself, but I don't set my foot into the jungle. Have you seen the size of these centipedes??"})
keywordHandler:addKeyword({"senja"}, StdModule.say, {npcHandler = npcHandler, text = "Um... I don't think so."})
keywordHandler:addKeyword({"svargrond"}, StdModule.say, {npcHandler = npcHandler, text = "I wish it was a little bit warmer there, but with a good mug of barbarian mead in your tummy everything gets a lot cosier."})
keywordHandler:addKeyword({"thalia"}, StdModule.say, {npcHandler = npcHandler, text = "I feel uncomfortable and rather unsafe in Thalia, so I don't really travel there."})
keywordHandler:addKeyword({"vega"}, StdModule.say, {npcHandler = npcHandler, text = "Um... I don't think so."})
keywordHandler:addKeyword({"venore"}, StdModule.say, {npcHandler = npcHandler, text = "Although it's the flourishing trade centre of Tibia, I don't like going there. Too much competition for my taste."})
keywordHandler:addKeyword({"time"}, StdModule.say, {npcHandler = npcHandler, text = "It's almost time to journey on."})
keywordHandler:addKeyword({"king"}, StdModule.say, {npcHandler = npcHandler, text = "Kings, queens, emperors and kaliphs... everyone claims to be different and unique, but actually it's the same thing everywhere."})

npcHandler:setMessage(MESSAGE_GREET, "Ah, a customer! Be greeted, |PLAYERNAME|!")
npcHandler:setMessage(MESSAGE_FAREWELL, "Farewell, |PLAYERNAME|, may the winds guide your way.")
npcHandler:setMessage(MESSAGE_WALKAWAY, "Come back soon!")
npcHandler:setMessage(MESSAGE_SENDTRADE, "Take all the time you need to decide what you want!")

local function onTradeRequest(cid)
	return true
end

npcHandler:setCallback(CALLBACK_ONTRADEREQUEST, onTradeRequest)
npcHandler:setCallback(CALLBACK_MESSAGE_DEFAULT, creatureSayCallback)
npcHandler:addModule(FocusModule:new())